File: 1346-check-if-n-and-its-double-exist.py
class Solution(object):
    def checkIfExist(self, arr):
        """
        :type arr: List[int]
        :rtype: bool
        """

        # Creating an empty dictionary
        dict1 = {}
        cnt = 0
        for x in arr:
            if(x != 0):
                if(x in dict1):
                    dict1[x] += 1
                else:
                    dict1[x] = 1
            else: cnt+=1
        if(cnt >= 2): return True
        for x in arr:
            if(x*2 in dict1): return True
        return False

File: 1346-check-if-n-and-its-double-exist.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        map<int, int> fre;
        int cnt = 0;
        for(auto x : arr)
            if(x != 0)
                fre[x] += 1;
            else 
                cnt+=1;
        if(cnt >= 2) return 1;
        for(auto x : arr)
            if(fre[x*2]>0) return 1;
        return 0;
    }
};
